### PR TITLE
Update kopia version to 02463ab

### PR DESCRIPTION
**Pull request by redgoat650**
## Change Overview

See https://github.com/kopia/kopia/commit/02463ab118c391c520995cc907dc070e944054cb

This PR updates the binary version in the kanister-tools image and the module version. It also updates the `golangci-lint` version in the build image and updates the build image to `v0.0.31`.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/kanisterio/kanister
 
-go 1.21.9
+go 1.22
+
+toolchain go1.22.0
```
Comment (julio-lopez) [inline go.mod:5]: why not 1.22.2 ?

File: .goreleaser.yml [hunk 2]
```diff
@@ -69,8 +69,8 @@ dockers:
   build_flag_templates:
   - "--pull"
   - "--build-arg=kan_tools_version={{ .Tag }}"
-# Refers to https://github.com/kopia/kopia/commit/5c85ca6bb087781eb3ea4044ffd87f0c00530c0a
-  - "--build-arg=kopia_build_commit=5c85ca6"
+# Refers to https://github.com/kopia/kopia/commit/02463ab118c391c520995cc907dc070e944054cb
+  - "--build-arg=kopia_build_commit=02463ab"
```
Comment (julio-lopez) [inline .goreleaser.yml:73]: kopia/kopia@1e98511c2ea07fbcfbe4cf0f29b58900bb32e460 ?

**Comment by redgoat650** [comment]
Closing in favor of https://github.com/kanisterio/kanister/pull/2848

File: .goreleaser.yml (hunk 2 above)
Comment (redgoat650) [reply .goreleaser.yml:73]: Latest commit updated in https://github.com/kanisterio/kanister/pull/2848

File: go.mod (hunk 1 above)
Comment (redgoat650) [reply go.mod:5]: Seems it was because my dev environment had go1.22.0 when I ran `go get`. Next PR uses 1.22.2 and updates the mod file accordingly

**Pull request closed without merging**